pkg/daemon/ceph/osd/volume.go

```diff
@@ -127,25 +129,38 @@ func getLVPath(op string) string {
 }
 
 func updateLVMConfig(context *clusterd.Context) error {
-	testArgs := []string{"-oL", "sed", "-i", "-e", "s#udev_sync = 1#udev_sync = 0#", "/etc/lvm/lvm.conf"}
-	if err := context.Executor.ExecuteCommand(false, "", "stdbuf", testArgs...); err != nil {
-		return fmt.Errorf("failed ceph-volume. %+v", err) // fail return here as validation provided by ceph-volume
+
+	if exists, _ := util.FileExists(lvmConfPath); !exists {
+		return fmt.Errorf("lvm conf file %s doesn't exist", lvmConfPath)
+	}
+
+	//disable udev
+	err := util.StringReplace(lvmConfPath, "udev_sync = 1", "udev_sync = 0", 1)
+	if err != nil {
+		return err
 	}
-	testArgs = []string{"-oL", "sed", "-i", "-e", "s#udev_rules = 1#udev_rules = 0#", "/etc/lvm/lvm.conf"}
-	if err := context.Executor.ExecuteCommand(false, "", "stdbuf", testArgs...); err != nil {
-		return fmt.Errorf("failed ceph-volume. %+v", err) // fail return here as validation provided by ceph-volume
+
+	err = util.StringReplace(lvmConfPath, "udev_rules = 1", "udev_rules = 0", 1)
+	if err != nil {
+		return err
 	}
-	testArgs = []string{"-oL", "sed", "-i", "-e", "s#use_lvmetad = 1#use_lvmetad = 0#", "/etc/lvm/lvm.conf"}
-	if err := context.Executor.ExecuteCommand(false, "", "stdbuf", testArgs...); err != nil {
-		return fmt.Errorf("failed ceph-volume. %+v", err) // fail return here as validation provided by ceph-volume
+
+	//disable lvmetad
+	err = util.StringReplace(lvmConfPath, "use_lvmetad = 1", "use_lvmetad = 0", 1)
+	if err != nil {
+		return err
 	}
-	testArgs = []string{"-oL", "sed", "-i", "-e", "s#scan = \\[ \"/dev\" \\]#scan = [ \"/dev\", \"/mnt\" ]#", "/etc/lvm/lvm.conf"}
-	if err := context.Executor.ExecuteCommand(false, "", "stdbuf", testArgs...); err != nil {
-		return fmt.Errorf("failed ceph-volume. %+v", err) // fail return here as validation provided by ceph-volume
+
+	//Change lv scanning to also look in /mnt
+	err = util.StringReplace(lvmConfPath, "scan = [ \"/dev\" ]", "scan = [ \"/dev\", \"/mnt\" ]", 1)
```

Comment:
instead of escaping the slashes, i'd suggest the backticks so it's more readable.
```
`scan = [ "/dev" ]`, `scan = [ "/dev", "/mnt" ]`
```

Reply:
updated